sdk/eventgrid/Azure.Messaging.EventGrid/tests/EventGridClientLiveTests.cs

```diff
@@ -62,33 +62,69 @@ public async Task CanPublishEventToDomain()
         [Test]
         public async Task CanPublishCloudEvent()
         {
-            EventGridClientOptions options = Recording.InstrumentClientOptions(new EventGridClientOptions());
-            EventGridClient client = InstrumentClient(
-                new EventGridClient(
+            EventGridPublisherClientOptions options = Recording.InstrumentClientOptions(new EventGridPublisherClientOptions());
+            EventGridPublisherClient client = InstrumentClient(
+                new EventGridPublisherClient(
                     new Uri(TestEnvironment.CloudEventTopicHost),
                     new AzureKeyCredential(TestEnvironment.CloudEventTopicKey),
                     options));
             await client.PublishCloudEventsAsync(GetCloudEventsList());
         }
 
+        [Test]
+        public async Task CanPublishCloudEventWithBinaryData()
+        {
+            EventGridPublisherClientOptions options = Recording.InstrumentClientOptions(new EventGridPublisherClientOptions());
+            EventGridPublisherClient client = InstrumentClient(
+                new EventGridPublisherClient(
+                    new Uri(TestEnvironment.CloudEventTopicHost),
+                    new AzureKeyCredential(TestEnvironment.CloudEventTopicKey),
+                    options));
+            await client.PublishCloudEventsAsync(GetCloudEventsListWithBinaryData());
```

Comment:
I'm thinking this pattern of defining a new GetEventWithX is a bit overkill as we need to create a new helper method for every new payload type, even when only using in a single test. It also makes it so we are not testing publishing events with different payload types in the same batch. I would say it would be better to just create the events in the test, and if we find that there are events that we are reusing across tests, use methods for those.